components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/SymmetricKeyInternalCryptoProvider.java

```diff
@@ -120,6 +127,17 @@ public SymmetricKeyInternalCryptoProvider(String secretKey) {
         }
     }
 
+    @Override
+    public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPIProvider,
+                          boolean returnSelfContainedCipherText) throws CryptoException {
+
+        byte[] encryptedKey = encrypt(cleartext, algorithm, javaSecurityAPIProvider);
+        //When symmetric encryption algorithm with padding is introduced, we can create self contained cipher texts.
+        //Then we can utilize the returnSelfContainedCipherText parameter and add the necessary logic to provide a
+        // selfcontained cipher text.
```

Comment:
Let's denote, as at now no cipher meta is included in the encryption data other than the cipher.

Reply:
Fixed with fde5a4d8773352de1654212bf563a4737bd658b2